### PR TITLE
chore: EKS custom iam example

### PR DESCRIPTION
**Pull request by varnastadeus**

### CODE REVIEW

File: examples/eks_cluster_custom_iam/eks.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  # Self Managed Node Group(s)
+  self_managed_node_group_defaults = {
+    vpc_security_group_ids = [aws_security_group.additional.id]
+  }
+
+  self_managed_node_groups = {
+    one = {
+      name          = "${var.cluster_name}-default"
+      instance_type = "t3.large"
```
[the remark this answers is not in the document]
Comment (varnastadeus) [reply examples/eks_cluster_custom_iam/eks.tf:35]: It is just an example, you can change it to whatever you want

File: examples/eks_cluster_custom_iam/eks.tf [hunk 2]
```diff
@@ -0,0 +1,78 @@
+module "eks" {
+  source       = "terraform-aws-modules/eks/aws"
+  version      = "18.21.0"
+  putin_khuylo = true
+
+  cluster_name    = var.cluster_name
+  cluster_version = "1.21"
```
Comment (r0kas) [inline examples/eks_cluster_custom_iam/eks.tf:7]: 1.22 is already available. Maybe bump it for example to look fresh for longer?

File: examples/eks_cluster_assumerole/main.tf [hunk 3]
```diff
@@ -59,9 +59,13 @@ module "castai-eks-cluster" {
   aws_assume_role_arn      = module.castai-eks-role-iam.role_arn
   aws_instance_profile_arn = module.castai-eks-role-iam.instance_profile_arn
 
-  subnets                  = var.subnets
-  override_security_groups = var.security_groups
-  tags                     = var.tags
+  subnets                  = module.vpc
```
Comment (aldor007) [inline examples/eks_cluster_assumerole/main.tf:62]:
```suggestion
  subnets                  = module.vpc.private_subnets
```

**Review by aldor007 — APPROVED** [review]

File: examples/eks_cluster_assumerole/main.tf (hunk 3 above)
Comment (varnastadeus) [reply examples/eks_cluster_assumerole/main.tf:62]: missed that, thx

File: examples/eks_cluster_custom_iam/iam.tf [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+      # Needed to tag non CAST nodes
+      {
+        Sid    = "NonResourcePermissions",
+        Effect = "Allow",
+        Action = [
+          "ec2:CreateTags",
+          # Necessary only if public SSH key for CAST nodes will be used
+          "ec2:ImportKeyPair",
+        ],
+        Resource = "*"
```
Comment (r0kas) [inline examples/eks_cluster_custom_iam/iam.tf:109]: Could we have a narrower scope here?

File: examples/eks_cluster_custom_iam/main.tf [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  subnets                  = module.vpc.private_subnets
+  override_security_groups = [
+    module.eks.cluster_security_group_id,
+    module.eks.node_security_group_id,
+    aws_security_group.additional.id,
+  ]
+  tags = var.tags
+
+  delete_nodes_on_disconnect = var.delete_nodes_on_disconnect
+}
```
Comment (r0kas) [inline examples/eks_cluster_custom_iam/main.tf:58]: newline

**Review by r0kas — APPROVED** [review]

File: examples/eks_cluster_custom_iam/iam.tf (hunk 4 above)
Comment (varnastadeus) [reply examples/eks_cluster_custom_iam/iam.tf:109]: yes and no, importKeyPair is optional anyway and createTags is not destructive action

**Pull request merged**